### PR TITLE
Threads

### PR DESCRIPTION
**Pull request by Vladimir-Bon-88**

### CODE REVIEW

File: src/main/java/edu/academy/jc/bondarenko/hw12/SumAccumulator.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    private int temp;
+    private int sum;
+
+    public SumAccumulator(List<Integer> list) {
+        this.list = list;
+    }
+
+    public void run() {
+        while (counter < 100) {
+            if (list.size() != 0) {
```
Comment (StanislavLapitsky) [inline src/main/java/edu/academy/jc/bondarenko/hw12/SumAccumulator.java:17]: list is out of synchronized block

Comment (Vladimir-Bon-88) [reply src/main/java/edu/academy/jc/bondarenko/hw12/SumAccumulator.java]: I changed. Check now please

File: src/main/java/edu/academy/jc/bondarenko/hw12/SumAccumulator.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            synchronized (list) {
+                if (list.size() != 0) {
+                    temp = list.get(0);
+                    list.remove(0);
+                    sum += temp;
+                    System.out.println("Sum is: " + sum);
+                    counter++;
+                    System.out.println("counter for accumulator " + counter);
+                } else {
+                    try {
```
Comment (StanislavLapitsky) [inline src/main/java/edu/academy/jc/bondarenko/hw12/SumAccumulator.java:26]: sleep() must be out of the synchronized block. In opposite case another thread does not have a chance to reach the list while the Accumulator is sleeping.

Comment (Vladimir-Bon-88) [reply src/main/java/edu/academy/jc/bondarenko/hw12/SumAccumulator.java]: Please, check now

**Review by StanislavLapitsky — APPROVED** [review]

**Pull request merged**